nexus/src/external_api/http_entrypoints.rs

```diff
@@ -7372,7 +7376,44 @@ async fn user_list(
     let handler = async {
         let opctx = OpContext::for_external_api(&rqctx).await?;
         let users = nexus
-            .silo_users_list_current(&opctx, &pagparams)
+            .silo_users_list_current(&opctx, &pagparams, &None)
+            .await?
+            .into_iter()
+            .map(|i| i.into())
+            .collect();
+        Ok(HttpResponseOk(ScanById::results_page(
+            &query,
+            users,
+            &|_, user: &User| user.id,
+        )?))
+    };
+    apictx.external_latencies.instrument_dropshot_handler(&rqctx, handler).await
+}
+
+/// List users
+#[endpoint {
+    method = GET,
+    path = "/v1/users",
+    tags = ["silos"],
+}]
+async fn user_list_v1(
+    rqctx: RequestContext<Arc<ServerContext>>,
+    query_params: Query<PaginatedById<params::OptionalGroupSelector>>,
+) -> Result<HttpResponseOk<ResultsPage<User>>, HttpError> {
+    let apictx = rqctx.context();
+    let handler = async {
+        let nexus = &apictx.nexus;
+        let query = query_params.into_inner();
+        let pagparams = data_page_params_for(&rqctx, &query)?;
+        let opctx = OpContext::for_external_api(&rqctx).await?;
+        let scan_params = ScanById::from_query(&query)?;
+        // TODO: a valid UUID gets parsed here and will 404 if it doesn't exist
+        // (as expected) but a non-UUID string just gets let through as None
+        // (i.e., ignored) instead of 400ing
```

Comment:
Hmmm, I'll have to dig into the underlying mechanism here. What we could do is add validation in `ScanById::from_query` that returns an error if the UUID is invalid. The flow here is a bit different though so I want to make sure I'm understanding it better.

Reply:
That’s a good point, we could do it semi-manually here. I was thinking it might need some serde attribute thing, but that would prevent the struct from being used in a place where we don’t want it to error out.

Reply:
Interestingly, this fixed it. The nested serde flatten was making it more lenient somehow.
https://github.com/oxidecomputer/omicron/pull/2423/commits/5402185764b0c208e15fb7ee11dd5d64b8052264#diff-aee01ef600c8eb547421397e5d89c41a67dd6e6ca6033676cac47b7909e1e344

Reply:
That is interesting. Once we're done with the V1 migration a next big step is wrapping up the selectors in a macro or something and potentially simplifying them. It won't result in an API change but I definitely feel like there's some improvement to be had there.